clients/venice-thin-client/src/main/java/com/linkedin/venice/client/stats/ClientStats.java

```diff
@@ -110,87 +106,141 @@ protected ClientStats(MetricsRepository metricsRepository, String storeName, Req
     Rate retryRequestSuccessKeyCount = new Rate();
     retryRequestSuccessKeyCountSensor =
         registerSensor("retry_request_success_key_count", retryRequestSuccessKeyCount, new Avg(), new Max());
-    retryKeySuccessRatioSensor = registerSensor(
-        "retry_key_success_ratio",
-        new TehutiUtils.SimpleRatioStat(retryRequestSuccessKeyCount, getSuccessRequestKeyCountRate()));
     multiGetFallbackSensor = registerSensor("multiget_fallback", new OccurrenceRate());
   }
 
   public void recordHttpRequest(int httpStatus) {
+    if (isMetaSystemStore()) {
```

Comment:
I feel like this is okay... probably there might be a better way of writing this, like creating a dummy blackhole sensor/metrics repo and in some internal implementation, check whether to direct to this blackhole sensor or not.
But if you think this is going to be more complicated, I think we can defer this effort. It is just that, for every new metrics getter setter in this class, we are going to repeat this if-else check for system store, and that does not super great to me.